.github/workflows/ci.yml

```diff
@@ -54,6 +54,26 @@ jobs:
       - name: check reminders and notify
         uses: agrc/reminder-action@v1
 
+  lockfiles:
+    runs-on: ubuntu-latest
+    name: "Lockfiles"
+    env: 
+      TARGET_DOCS_LOCKFILE_PATH: .lockfiles/py310-dev.lock
+      DOCS_LOCKFILE_PATH: /tmp/lockfile
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        id: setup-python
+        with:
+          python-version: '3.10'
+      - name:  Verify lockfile integrity 
+        run: |
+          pip install tox-uv
+          tox -e docs-lockfile 
+          if ! diff --ignore-matching-lines="^\#" $DOCS_LOCKFILE_PATH $TARGET_DOCS_LOCKFILE_PATH >/dev/null; then
+            echo "Lockfile needs to be updated"
```

Comment:
whats the design idea? w have to update lockfiels manually? any feeling for how often that will happen? Can iamgine that will trigger allt he time invaldiating all pipelines all the time. Do you want the PR owner to update the lockgile then?

An alternative could be to not let the step fail but print a warning like
```bash
echo "::warning file=app.py,line=10,col=5::Code formatting issues detected"
```
![image](https://github.com/user-attachments/assets/08bcfe24-2ca0-4bcd-b248-0f285b605401)

Reply:
Totally open with any ideas here. The workflow is absolutely not set in stone, it's just a rough first draft with the bare minimum logic.

But to answer the question: it is basically thought as a consistency check to remind you updating the lockfile, similar to the changelog reminder, but currently with the idea of letting it fail because if there is a diff .... well then it basically means you are no longer guaranteed that you'll be able to rebuild the docs later. Don't know if there is a better alternative or how people handle these lockfiles in general in their repos/ci/workflows. Pretty much my first touchpoint.

> Can iamgine that will trigger allt he time invaldiating all pipelines all the time
If everything works as I hope, we would only get the warning possibly when pyproject.toml has changed. If that is done, a single `tox -e docs-lockfile` will update the file in a matter of a few secs

Reply:
I guess its not a problem if we shift it to dev responsibility
I always perofrm tox mypy and lint before pushing alrge chunks, so this would siply ebcome another one and if it wouldt change theres also nothig to commit
if it were changing then we are warned by CI via error

seems OK to me lets try?

Reply:
exactly, that was the idea 👍🏼